image-upload

```diff
@@ -95,21 +109,28 @@ def main():
         sources.append(source)
 
     for source in sources:
+        basename = os.path.basename(source)
+        public = not any(basename.startswith(pfx) for pfx in ('rhel', 'windows'))
+
         # determine possible stores, unless explicitly given
         stores = args.store
         if not stores:
+            stores = list(HYBRID_STORES)
+
             # these images are not freely redistributable, keep them Red Hat internal
-            b = os.path.basename(source)
-            if b.startswith("rhel") or image.startswith("windows"):
-                stores = REDHAT_STORES
+            if public:
+                stores.extend(PUBLIC_STORES)
             else:
-                stores = PUBLIC_STORES
+                stores.extend(REDHAT_STORES)
 
+        success = False
         for store in stores:
-            ret = upload(store, source)
+            dest = urllib.parse.urljoin(store, basename)
+            ret = upload(dest, source, public)
             if ret == 0:
-                return ret
-        else:
+                success = True
```

Comment:
This subtle change has the effect that image-upload will  now upoad to *all* stores. I am a bit afraid that this will take rather long for our AWS mirror. But we want to replace these "legacy stores" anyway as soon as the dust settles on this, so until then this is a good enough quick fix. Better that than accidentally locking ourselves out of the image stores.